libdnf/base/base.cpp

```diff
@@ -64,10 +64,12 @@ Base * Base::get_locked_base() noexcept {
     return locked_base;
 }
 
-void Base::load_config_from_file(const std::string & path) {
+void Base::load_config_from_file(const std::string & path) try {
     ConfigParser parser;
     parser.read(path);
     config.load_from_parser(parser, "main", vars, *get_logger());
+} catch (const Error & e) {
+    std::throw_with_nested(RuntimeError(M_("Unable to load configuration file \"{}\""), path));
```

Comment:
`RuntimeError` should be gradually removed, you should use a specific error class instead. But it seems there isn't one at the moment and it's a bit awkward (with `load_from_parser()` throwing an error but not having the file name), so feel free to keep it, maybe add a TODO?

Reply:
I can add "TODO", but we were talking about replacing `RuntimeError`s in the future. Are we going to add "TODO" to all the `RuntimeError`? I don't know how much easier it will be to search for "TODO" instead of "RuntimeError".

Reply:
Yeah, having `RuntimeError` there is enough. I just thought you'd add some description with the TODO but it's not necessary.